src/Parameters.jl

```diff
@@ -460,4 +460,25 @@ macro pack(arg)
     )
 end
 
+"""
+Splats keys from a dict into variables
+
+```
+@materialize a, b, c = dict
+```
+
+"""
+macro materialize(dict_splat)
+    keynames, dict = dict_splat.args
+    keynames = isa(keynames, Symbol) ? [keynames] : keynames.args
+    dict_instance = gensym()
```

Comment:
Oh sorry! Didn't see this... Yeah, that's for the case when the dict gets computed (which you don't want to do for every key access)!
I've actually never used the macro like that so far, though :D